tests/0045-subscribe_update.c

```diff
@@ -821,6 +821,107 @@ static void do_test_resubscribe_with_regex() {
         SUB_TEST_PASS();
 }
 
+/**
+ * @brief Create many topics and apply several subscription
+ *        updates, unsubscribing and re-subscribing too.
+ *        After changing some subscriptions verifies that the assignment
+ *        corresponds to last one.
+ *
+ * @param with_rebalance_cb Use a rebalance callback to perform the assignment.
+ *                          It needs to poll the consumer when awaiting for the
+ *                          assignment in this case.
+ */
+static void do_test_subscribe_many_updates(rd_bool_t with_rebalance_cb) {
+#define TOPIC_CNT 100
+        char *topics[TOPIC_CNT] = {0};
+        char *topic;
+        size_t i;
+        char *group;
+        rd_kafka_t *rk;
+        rd_kafka_conf_t *conf;
+        const int max_subscription_size = 5, partition_cnt = 4;
+        rd_kafka_topic_partition_list_t *expected_assignment = NULL;
+
```

Comment:
Should be move inside the topic for loop as this is not needed outside.

Reply:
no because that way it's reinitialized to NULL at every iteration.